inspirehep/modules/workflows/utils.py

```diff
@@ -141,3 +145,67 @@ def _try_to_log(logfn, *args, **kwargs):
         return res
 
     return _decorator
+
+
+@with_debug_logging
+def get_pdf_in_workflow(obj):
+    """Return the fullpath to the PDF attached to a workflow object
+
+    Return the PDF attached to a workflow object. If no PDF is already attached
+    and the workflow object is related to an arXiv record, fetches on the fly
+    the PDF from arXiv and attaches it.
+    """
+    for filename in obj.files.keys:
+        if filename.endswith('.pdf'):
+            # There's a PDF already attached
+            return obj.files[filename].file.uri
+
+    arxiv_id = get_arxiv_id(obj.data)
+    submission_pdf = obj.extra_data.get('submission_pdf')
+    if arxiv_id:
+        # It's an arXiv record, we give the PDF from arXiv
+        filename = secure_filename("{0}.pdf".format(arxiv_id))
+        # We fetch if on the fly if needed
+        pdf = download_file_to_workflow(
+            workflow=obj,
+            name=filename,
+            url=current_app.config['ARXIV_PDF_URL'].format(
+                arxiv_id=arxiv_id
+            )
+        )
+        pdf['doctype'] = "arXiv"
+        obj.log.info("PDF retrieved from arXiv for {0}".format(arxiv_id))
+        return obj.files[filename].file.uri
+    elif submission_pdf:
+        # It's a user submission and the user has provided the URL for a PDF.
+        filename = secure_filename("fulltext.pdf")
+        pdf = download_file_to_workflow(
+            workflow=obj,
+            name=filename,
+            url=submission_pdf
+        )
+        if pdf:
+            pdf['doctype'] = 'main'
+            obj.log.info("PDF provided by user from {0}".format(submission_pdf))
+            return obj.files[filename].file.uri
```

Comment:
Maybe then we should have another util that checks if a `WorkflowObject` has a PDF attached, instead of calling a util that does that check AND attempts to fetch the PDF if it's not there. I mean, functions should one thing well.